### DISCUSSION TITLE
MetricTable: Serialization of Closure not allowed

### DISCUSSION THREAD

**Discussion by daanadriaan** [laravel/nova-issues#4598] [Ideas & Feature Requests] [answered]
- Laravel Version: 9
- Nova Version: 4
- PHP Version: 8

### Description:
MetricTable's don't support cache options. Would be nice if the documentation would explain this, or even nicer if this would be possible.

**Comment by juse-less** [comment]
I reckon the MetricTable doesn't have it.. but.. do you happen to use the `transform()` method on the Metric?
I know from any other Metric, that we use `transform()` on, will throw this error if cache is also being used.
The solution to `transform()` is to transform the value as part of the database query, or disable caching.
I haven't found another (or better) way.

So, with that in mind
* Try figuring out which method(s), where Closures are used, that would cause it.
  Some methods can be omitted by doing things elsewhere (like the database query).
* Another thing to think of - does your MetricTable really need to be cached?
  Normally it'd be heavier database queries that one would want to cache.
  Like in the other Metrics where aggregates, maybe plenty of references across many tables, etc, are being used.
  If you don't need to use cache, disable it and run `php artisan optimize:clear`, or at least `php artisan cache:clear`.

Serialising Closures is a bit hackish, and not directly supported at language level.
There are some packages that can do it. But again, it's hackish and could cause issues.

I recall @crynobone answering something like this in another discussion or issue.

**Comment by crynobone** [comment, marked as answer]
This has been implemented in https://nova.laravel.com/releases/4.12.8
